lib/engine-addon.js

```diff
@@ -126,9 +129,17 @@ var buildVendorCSSTree = memoize(function buildVendorCSSTree(vendorTree) {
 
 var buildEngineJSTree = memoize(function buildEngineJSTree() {
   var engineSourceTree;
-  var treePath = path.resolve(this.root, this.treePaths['addon']);
-  if (existsSync(treePath)) {
+  var treePath;
+  var addonTree = this.options.trees.addon;
```

Comment:
The key is that we expect a tree not a path.  `treePaths` is for how we represent "where" things are, but this is specifically to allow custom filtering and whatnot of the tree.